verifier/ita/client.go

```diff
@@ -0,0 +1,250 @@
+package ita
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha512"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-tpm-tools/verifier"
+)
+
+const (
+	nonceEndpoint = "/appraisal/v2/nonce"
+	tokenEndpoint = "/appraisal/v2/attest/gcp/confidentialspace"
+
+	apiKeyHeader      = "x-api-key"
+	acceptHeader      = "Accept"
+	contentTypeHeader = "Content-Type"
+	applicationJSON   = "application/json"
+
+	challengeNamePrefix = "ita://"
+)
+
+var regionalURLs map[string]string = map[string]string{
+	"US": "https://api.trustauthority.intel.com",
+	"EU": "https://api.eu.trustauthority.intel.com",
+}
+
+type client struct {
+	inner  *http.Client
+	apiURL string
+	apiKey string
+}
+
+func urlFromRegion(region string) (string, error) {
+	if region == "" {
+		return "", errors.New("API region required to initialize ITA client")
+	}
+
+	url, ok := regionalURLs[strings.ToUpper(region)]
+	if !ok {
+		// Create list of allowed regions.
+		keys := []string{}
+		for k := range regionalURLs {
+			keys = append(keys, k)
+		}
+		return "", fmt.Errorf("unsupported region %v, expect one of %v", region, keys)
+	}
+
+	return url, nil
+}
+
+func NewClient(region string, key string) (verifier.Client, error) {
+	url, err := urlFromRegion(region)
+	if err != nil {
+		return nil, err
+	}
+
+	return &client{
+		inner: &http.Client{
+			Transport: &http.Transport{
+				// https://github.com/intel/trustauthority-client-for-go/blob/main/go-connector/token.go#L130.
+				TLSClientConfig: &tls.Config{
+					CipherSuites: []uint16{
+						tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+						tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+					},
+					InsecureSkipVerify: false,
+					MinVersion:         tls.VersionTLS12,
+				},
+				Proxy: http.ProxyFromEnvironment,
+			},
+		},
+		apiURL: url,
+		apiKey: key,
+	}, nil
+}
+
+// Confirm that client implements verifier.Client interface.
+var _ verifier.Client = (*client)(nil)
+
+type itaNonce struct {
+	Val       []byte `json:"val"`
+	Iat       []byte `json:"iat"`
+	Signature []byte `json:"signature"`
```

Comment:
Is verifying the Signature needed? Maybe in createHashedNonce

Reply:
We don't need to verify the signature - this should be done by ITA. We should - however - pass the nonce object back to ITA when requesting a token.